js/ui/windowManager.js

```diff
@@ -1059,29 +1074,51 @@ const WindowManager = new Lang.Class({
             return;
         }
 
-        let origX = actor.x;
-        if (origX == monitor.x) {
-            // the side bar will appear from the left side
-            actor.set_position(monitor.x - actor.width, actor.y);
-        } else {
-            // ... from the right side
-            actor.set_position(monitor.x + monitor.width, actor.y);
+        if (SideComponent.isGrandCentralWindow(actor.meta_window)) {
+            // the GrandCentral window will appear from the top center
+            let origY = actor.y;
+            actor.set_position(actor.x, monitor.y - actor.height);
+
+            actor.opacity = 255;
+            actor.show();
+
+            Tweener.addTween(actor,
+                             { y: origY,
+                               time: WINDOW_ANIMATION_TIME,
+                               transition: "easeOutQuad",
+                               onComplete: this._mapWindowDone,
+                               onCompleteScope: this,
+                               onCompleteParams: [shellwm, actor],
+                               onOverwrite: this._mapWindowOverwrite,
+                               onOverwriteScope: this,
+                               onOverwriteParams: [shellwm, actor]
+                             });
         }
+        else {
+            let origX = actor.x;
+            if (origX == monitor.x) {
+                // the side bar will appear from the left side
+                actor.set_position(monitor.x - actor.width, actor.y);
+            } else {
+                // ... from the right side
+                actor.set_position(monitor.x + monitor.width, actor.y);
+            }
 
-        actor.opacity = 255;
-        actor.show();
+            actor.opacity = 255;
+            actor.show();
```

Comment:
They can probably be called on line 1085 though - it shouldn't be a problem if they are called after `Tweener.addTween` since that is asynchronous anyway.